test/feature/middleware_support.feature

```diff
@@ -0,0 +1,153 @@
+Feature: Middleware support
+
+  Scenario Outline: simple middleware support
+    Given a file "routes.coffee" with the content
+      """
+      module.exports = ({ GET }) ->
+        GET '/users', to: 'users#index'
+      """
+    And a file "controllers/users_controller.coffee" with the content
+      """
+      middle = (req, res, next) ->
+        req.custom = 'foo'
+        next()
+
+      class UsersController
+        middleware:
+          index: middle
+
+        index:   (req, res) -> res.end req.custom
```

Comment:
Cool idea @alexpeachey!

I think a more realistic use case would help motivate this feature better, i.e. lay out why this complexity should be added. Is there something super simple that is typically done in controllers that should be extracted like this? 

Also, this seems to be more a `before` hook for Expressive controllers than a middleware (which already has a meaning in express, and which runs before and after requests: http://expressjs.com/guide/using-middleware.html). Maybe adjust the name?